packages/framework/src/Model/Seo/SeoSettingFacade.php

```diff
@@ -59,37 +54,55 @@ public function getDescriptionsMainPageIndexedByDomainIds(array $domainConfigs)
 
     /**
      * @param int $domainId
-     * @return string
+     * @return string|null
      */
-    public function getTitleAddOn($domainId)
+    public function getTitleAddOn(int $domainId): ?string
     {
         return $this->setting->getForDomain(self::SEO_TITLE_ADD_ON, $domainId);
     }
 
     /**
-     * @param string $value
      * @param int $domainId
+     * @return string|null
      */
-    public function setTitleMainPage($value, $domainId)
+    public function getRobotsTxtContent(int $domainId): ?string
+    {
+        return $this->setting->getForDomain(self::SEO_ROBOTS_TXT_CONTENT, $domainId);
+    }
+
+    /**
+     * @param string|null $value
+     * @param int $domainId
+     */
+    public function setTitleMainPage(?string $value, int $domainId): void
     {
         $this->setting->setForDomain(self::SEO_TITLE_MAIN_PAGE, $value, $domainId);
     }
 
     /**
-     * @param string $value
+     * @param string|null $value
      * @param int $domainId
      */
-    public function setDescriptionMainPage($value, $domainId)
+    public function setDescriptionMainPage(?string $value, int $domainId): void
     {
         $this->setting->setForDomain(self::SEO_META_DESCRIPTION_MAIN_PAGE, $value, $domainId);
     }
 
     /**
-     * @param string $value
+     * @param string|null $value
      * @param int $domainId
      */
-    public function setTitleAddOn($value, $domainId)
+    public function setTitleAddOn(?string $value, int $domainId): void
```

Comment:
adding return type and typehint is a BC-break. please remove it.

Reply:
Removed.